### PR TITLE
feat: add checking of user existence in DB in auth Middleware and fetchMe Rest Api for User (#40012)

### PR DESCRIPTION
**Pull request by permil-successive**
@Prabal555 @nxvinay

### CODE REVIEW

**Comment by permil-successive** [comment]
token

eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwiZW1haWwiOiJ2aW5heUBzdWNjZXNzaXZlLnRlY2giLCJpc3MiOiJTdWNjZXNzaXZlIFRlY2hub2xvZ2llcyIsImlkIjoiNWUzZDFlNmNjYmVhYmU2NmMxNmQyNzBiIiwiaWF0IjoxNTE2MjM5MDIyfQ.SIe_hxuvNjoH5PA7ZDv10C_vsHcVm5JSr9PCfLCbGpU

**Comment by permil-successive** [comment]
db document

{
    "_id" : "5e3d1e6ccbeabe66c16d270b",
    "hobbies" : [ 
        "Traveling"
    ],
    "name" : "Naveen",
    "role" : "head-trainer",
    "address" : "Pune",
    "dob" : ISODate("1992-12-31T18:30:00.000Z"),
    "email" : "vinay@successive.tech",
    "mobileNumber" : 1231231230,
    "__v" : 0
}

File: src/controllers/trainee/routes.ts [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
-.get(list)
-.post(create)
-.put(update)
-.delete(deleteFunction);
+.get(authorizationHandler(moduleName, PermissionType.read), validationHandler(getConfig), list)
+.post(authorizationHandler(moduleName, PermissionType.write), validationHandler(createConfig), create)
+.put(authorizationHandler(moduleName, PermissionType.write), validationHandler(updateConfig), update)
+.delete(authorizationHandler(moduleName, PermissionType.delete), validationHandler(deleteConfig), deleteFunction);
+
+traineeRoutes.delete('/:id', authorizationHandler('trainee', 'delete'), validationHandler(validationConfig.delete), deleteFunction);
```
Comment (Prabal555) [inline src/controllers/trainee/routes.ts]: are two delete method intentional

File: src/controllers/trainee/validations.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      number: true,
+      in: ['query'],
+      errorMessage: 'skip is invalid',
+    },
+    limit: {
+      required: false,
+      default: 10,
+      number: true,
+      in: ['query'],
+      errorMessage: 'imit is invalid',
```
Comment (Prabal555) [inline src/controllers/trainee/validations.ts]: typo

File: src/controllers/user/routes.ts [hunk 3]
```diff
@@ -1,13 +1,20 @@
 import { Router } from 'express';
 import Controller from './controller';
+import validationConfig from './validations';
+import { validationHandler, authorizationHandler, PermissionType } from '../../libs';
 
 const userRoutes: Router  = Router();
-const { list, create, update, delete: deleteFunction } = Controller;
+const { list, create, update, delete: deleteFunction, fetchMe } = Controller;
+const { get: getConfig, create: createConfig, update: updateConfig, delete: deleteConfig } = validationConfig;
+const moduleName = 'user';
```
Comment (Prabal555) [inline src/controllers/user/routes.ts]: the module name will not work, use 'users' instead

File: src/controllers/trainee/validations.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        }
+
+        return true;
+      }
+    },
+    name: {
+      required: true,
+      regex: /^[a-z ,.'-]+$/i,
+      in: ['body'],
+      errorMessage: 'Name is required',
```
Comment (Prabal555) [inline src/controllers/trainee/validations.ts]: validations don not cover the entire JSON object that is used for creation

**Comment by Prabal555** [comment]
Please take care of the following issues: 

1. there should be some unique key in your create object.

**Review by Prabal555 — CHANGES_REQUESTED** [review]

File: src/controllers/trainee/validations.ts (hunk 2 above)
Comment (Prabal555) [inline src/controllers/trainee/validations.ts]: limit and skip does not work on the api

**Review by Prabal555 — CHANGES_REQUESTED** [review]

**Comment by Prabal555** [comment]
Please add a check in the for ids not available for update and delete.

File: src/controllers/trainee/routes.ts (hunk 1 above)
Comment (permil-successive) [reply src/controllers/trainee/routes.ts]: deleted unwanted method

**Pull request merged**